ev3dev2/motor.py

```diff
@@ -890,8 +941,8 @@ def on_for_rotations(self, speed, rotations, brake=True, block=True):
         ``speed`` can be a percentage or a :class:`ev3dev2.motor.SpeedValue`
         object, enabling use of other units.
         """
-        if speed is None or rotations is None:
-            raise ValueError("Either speed ({}) or rotations ({}) is None".format(self, speed, rotations))
+        assert speed, "{}: speed is invalid ({}), motor will not move".format(self, speed)
```

Comment:
Not sure what you are asking here?

Reply:
If speed is zero it looks like this assertion will fail; previously if speed was zero the error _would not_ be thrown. Same with rotations.

Reply:
Right but this is part of what we discussed yesterday, we were all over the place how to handle 0 values for speed, etc. We did a mix of
- used an assert
- logged a warning  and stopped the motors
- not check at all for 0 values

I thought we agreed to use asserts so I made them all consistent to use asserts?

Reply:
Zero speed/distance are a supported scenario; there was a fair amount of discussion on the topic in https://github.com/ev3dev/ev3dev-lang-python/pull/523.

Reply:
What is the desired behavior...log a warning and return? I’m fine either way we just need to make it consistent

Reply:
The existing tests document the expected behavior in all scenarios. In general a `0` causes a no-op — it's fine to ask to go nowhere. No error is logged because it isn't a failing scenario.